Security/Http/Authentication/AuthenticationFailureHandler.php

```diff
@@ -33,13 +34,29 @@ public function __construct(EventDispatcherInterface $dispatcher)
     public function onAuthenticationFailure(Request $request, AuthenticationException $exception)
     {
         $errorMessage = strtr($exception->getMessageKey(), $exception->getMessageData());
+        $statusCode = self::mapExceptionCodeToStatusCode($exception->getCode());
+
         $event = new AuthenticationFailureEvent(
             $exception,
-            new JWTAuthenticationFailureResponse($errorMessage)
+            new JWTAuthenticationFailureResponse($errorMessage, $statusCode)
         );
 
         $this->dispatcher->dispatch($event, Events::AUTHENTICATION_FAILURE);
 
         return $event->getResponse();
     }
+
+    /**
+     * @param string|int $exceptionCode
+     */
+    private static function mapExceptionCodeToStatusCode($exceptionCode): int
```

Comment:
```suggestion
    private static function mapExceptionCodeToStatusCode(int $exceptionCode): int
```
seems safe

Reply:
I'm mitigated: on one hand you are right it can only be an `int`, on the other `::getCode()` is coming from `Exception` which technically can be a string e.g. for `PDOException`...

Reply:
fair enough, let’s keep it as-is